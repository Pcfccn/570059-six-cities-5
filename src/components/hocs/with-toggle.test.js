import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withToggle from "./with-toggle/with-toggle";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withToggle(MockComponent);

it(`withToggle is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped>
      <React.Fragment />
    </MockComponentWrapped>
  )).toJSON();

  expect(tree).toMatchSnapshot();
});
