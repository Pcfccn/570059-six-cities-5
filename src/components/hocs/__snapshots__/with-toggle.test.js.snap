// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withToggle is rendered correctly 1`] = `<div />`;
